lib/internal/url.js

```diff
@@ -958,15 +958,11 @@ class URL {
     if (protocol === 'blob:') {
       const path = this.pathname;
       if (path.length > 0) {
-        try {
-          const out = new URL(path);
-          // Only return origin of scheme is `http` or `https`
-          // Otherwise return a new opaque origin (null).
-          if (out.#context.scheme_type === 0 || out.#context.scheme_type === 2) {
-            return `${out.protocol}//${out.host}`;
-          }
-        } catch {
-          // Do nothing.
+        const out = URL.parse(path);
+        // Only return origin of scheme is `http` or `https`
+        // Otherwise return a new opaque origin (null).
+        if (out && (out.#context.scheme_type === 0 || out.#context.scheme_type === 2)) {
```

Comment:
This won't work because apparently private properties with `#` doesn't work with optional chaining.

TIL.

Reply:
How does it even work at all? I mean, how are we allowed to do `out.#context` in the first place, since `#context` is private?

Reply:
> This won't work because apparently private properties with `#` doesn't work with optional chaining.
> 
> TIL.

I just tested this, and AFAICT they do:
```javascript
class SomePrivateClass {
    #privateField = {
        value: 42
    }

    getPrivateField() {
        return this.#privateField?.value;
    }
}

const somePrivateInstance = new SomePrivateClass();
console.log(somePrivateInstance.getPrivateField()); // 42
```

But even so, if we arent in the class (like @targos said), how would this work at all?

Reply:
Apparently they do, my bad.

```
> class SomePrivateClass {
...     #privateField = {
...         value: 42
...     }
...
...     getPrivateField() {
...         return this?.#privateField?.value;
...     }
... }
undefined
> let instance = new SomePrivateClass()
undefined
> instance.getPrivateField()
42
```

Reply:
@targos This is inside a function defined in URL class.

Reply:
> @targos This is inside a function defined in URL class.

Oh! I didn't know that would work, that's pretty interesting (with the getting of a different instance):
```js
class SomePrivateClass {
    #privateField = {
        value: 42
    }

    getPrivateField() {
        return this.#privateField?.value;
    }

    getSelfPrivateField(otherInstance) {
        return otherInstance.#privateField?.value;
    }
}

const somePrivateInstance = new SomePrivateClass();
console.log(somePrivateInstance.getPrivateField()); // 42
const someOtherPrivateInstance = new SomePrivateClass();
console.log(somePrivateInstance.getSelfPrivateField(someOtherPrivateInstance)); // Also 42
```

Reply:
> How does it even work at all? I mean, how are we allowed to do `out.#context` in the first place, since `#context` is private?

TIL:

```js
class Test {
  #priv = 'yep'

  getPriv(obj) {
    return obj.#priv
  }
}

class Test2 {
  #priv = 'nope'

  getPriv(obj) {
    return obj.#priv
  }
}

const test = new Test();
const testOther = new Test();
const test2 = new Test2();

console.log(test.getPriv(test))
console.log(test.getPriv(testOther))
console.log(test.getPriv(test2))
```

```
> node test.js
yep
yep
D:\Git\test.js:5
    return obj.#priv
               ^

TypeError: Cannot read private member #priv from an object whose class did not declare it
    at Test.getPriv (D:\Git\test.js:5:16)
```